bundles/core/org.eclipse.smarthome.core.test/src/test/groovy/org/eclipse/smarthome/core/service/AbstractWatchServiceTest.groovy

```diff
@@ -0,0 +1,158 @@
+import org.junit.After
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test for {@link AbstractWatchService}.
+ * 
+ * @author Dimitar Ivanov
+ *
+ */
+class AbstractWatchServiceTest extends OSGiTest {
+
+    def static ROOT = "testRoot" + File.separatorChar + "files"
+
+    RelativeWatchService relativeService
+
+    @BeforeClass
+    static void setUpBeforeClass(){
+        File root = new File(ROOT);
+        root.mkdirs()
+    }
+
+    @AfterClass
+    static void tearDownClass(){
+        File root = new File(ROOT);
+        root.deleteDir()
+    }
+
+    @Before
+    public void setUp(){
+        relativeService = new RelativeWatchService(ROOT)
+        relativeService.activate()
+    }
+
+    @After
+    public void tearDown(){
+        relativeService.deactivate()
+    }
+
+    @Test
+    void 'AbstractWatchQueueReader processWatchEvent path provided with correct arguments'() {
+        // File created
+        def fileName = "watchFile"
+        File watchFile1 = new File(ROOT,fileName)
+        watchFile1.createNewFile()
+
+        fullEventAssertionsByKind(fileName,ENTRY_CREATE,false)
+
+        // File modified
+        watchFile1 << "Additional content"
+        fullEventAssertionsByKind(fileName,ENTRY_MODIFY,false)
+
+        // File deleted
+        boolean isDeleted = watchFile1.delete()
+        assertThat "Test file is not deleted", isDeleted,is(true)
+        fullEventAssertionsByKind(fileName,ENTRY_DELETE,true)
+    }
+
+    void fullEventAssertionsByKind(fileName, kind, osSpecific){
+        // Wait one second for any change to be detected by the watch service
+        sleep 1000
+
+        // As the implementation of the watch events is OS specific, it can happen that when the file is deleted two events are fired - ENTRY_MODIFY followed by an ENTRY_DELETE
+        // This is usually observed on Windows and below is the workaround
+        // Related discussion in StackOverflow: http://stackoverflow.com/questions/28201283/watchservice-windows-7-when-deleting-a-file-it-fires-both-entry-modify-and-e
+        boolean isDeletedWithPrecedingModify = osSpecific && kind.equals(ENTRY_DELETE) && relativeService.allFullEvents.size() == 2 && relativeService.allFullEvents[0].eventKind.equals(ENTRY_MODIFY)
+        if(isDeletedWithPrecedingModify){
+            // Remove the ENTRY_MODIFY element as it is not needed
+            relativeService.allFullEvents.remove(0)
+        }
+
+        waitForAssert {assertThat "Only one event for file $fileName is expected shortly after the file has been altered. Here are the found events: " + relativeService.allFullEvents, relativeService.allFullEvents.size(), is(1)}
+        FullEvent fullEvent = relativeService.allFullEvents[0]
+
+        assertThat "The path of the processed event should be relative to the root", fullEvent.eventPath.toString(), is(ROOT + File.separatorChar + fileName)
```

Comment:
You are completely right, the path now is not relative, but absolute. I found out that I have to do some refactoring in order to make it build the relative path, so soon I will update the PR and also add some more tests for testing the relative path accordingly. Thanks for the remark!